src/main/resources/hudson/security/GlobalMatrixAuthorizationStrategy/config.jelly

```diff
@@ -202,20 +202,64 @@ THE SOFTWARE.
           e.onclick = function() {
             var tr = findAncestor(this,"TR");
             var inputs = tr.getElementsByTagName("INPUT");
-            for(var i=0; i&lt;inputs.length; i++){
+            for(var i=0; i < inputs.length; i++){
                 if(inputs[i].type == "checkbox") inputs[i].checked = false;
             }
+            Behaviour.applySubtree(findAncestor(this,"TABLE"),true);
             return false;
           };
           e = null; <!-- avoid memory leak -->
         });
+      Behaviour.specify("#${strategyid} td input", 'GlobalMatrixAuthorizationStrategy', 0, function(e) {
+        var impliedByString = findAncestor(e, "TD").getAttribute('data-implied-by-list');
+        var impliedByList = impliedByString.split(" ");
+        var tr = findAncestor(e,"TR");
+        e.disabled = false;
+        for (var i = 0; i < impliedByList.length; i++) {
+          var permissionId = impliedByList[i];
+          var reference = tr.querySelector("td[data-permission-id='" + permissionId + "'] input");
+          if (reference !== null) {
+            if (reference.checked) {
+              e.disabled = true;
+            }
+          }
+        }
+        e.onchange = function() {
+          var permissionId = findAncestor(this, "TD").getAttribute('data-permission-id');
+          var checked = this.checked;
+          var tr = findAncestor(this,"TR");
+          var inputs = tr.getElementsByTagName("INPUT");
+          for (var i=0; i < inputs.length; i++){
+            if(inputs[i].type == "checkbox") {
+              var td = findAncestor(inputs[i], "TD");
+              if (td !== null) {
+                var list = td.getAttribute('data-implied-by-list');
+                if (list.includes(permissionId)) {
+                  inputs[i].disabled = checked;
+                }
+              }
+            }
+          }
+          return true;
+        };
+        e = null; <!-- avoid memory leak -->
```

Comment:
No idea. It's been in Git since the beginning, https://github.com/jenkinsci/jenkins/blob/1.312/core/src/main/resources/hudson/security/GlobalMatrixAuthorizationStrategy/config.jelly#L146, i.e. it's at least 10 years old. Perhaps related to browser bugs from a decade ago?